src/services/image.service.ts

```diff
@@ -0,0 +1,34 @@
+import type { Playlist, PlaylistItem } from '#types/playlist';
+import { findPlaylistImageForWidth } from '#src/utils/collection';
+import type { Config } from '#types/Config';
+
+export type ImageProperty = 'shelfImage' | 'backgroundImage' | 'channelLogoImage';
+
+export const generateAlternateImageURL = (mediaid: string, imageLabel: string, width: number) =>
+  `https://img.jwplayer.com/v1/media/${mediaid}/images/${imageLabel}.jpg?width=${width}`;
+
+export const getAlternateImageLabel = (config: Config, propertyName: ImageProperty, item: PlaylistItem, playlist?: Playlist) => {
+  const alternateImageLabel = propertyName === 'shelfImage' ? playlist?.[propertyName] : item[propertyName];
+  const configImageLabel = config.custom?.[propertyName];
+
+  // image label override from playlist or playlist item
+  if (typeof alternateImageLabel === 'string') {
+    return alternateImageLabel;
+  }
+
+  // image label override from config
+  if (typeof configImageLabel === 'string') {
+    return configImageLabel;
+  }
+};
+
+export const getMediaImages = (config: Config, propertyName: ImageProperty, item: PlaylistItem, playlist?: Playlist, width = 640) => {
+  const posterImage = findPlaylistImageForWidth(item, width);
+  const alternateImageLabel = getAlternateImageLabel(config, propertyName, item, playlist);
+
+  if (alternateImageLabel) {
+    return [generateAlternateImageURL(item.mediaid, alternateImageLabel, width), posterImage];
+  }
+
+  return [posterImage];
```

Comment:
I think it's nicer to have named properties returned, otherwise there's no real way to know if image[0] is the main or the fallback except by inspecting the code and deciphering how it's used.  Note, it doesn't hurt to have fallback hardcoded as poster, but I'd be find returning undefined for it if it's already set as the main image.

I would also suggest we use this type for `image` throughout the application so we aren't adding extra props (i.e. having both `image` and `fallbackImage`) to everything.
```suggestion
  return {
    image: alternateImageLabel ? generateAlternateImageURL(item.mediaid, alternateImageLabel, width) : posterImage,
    fallbackImage: posterImage
  };
```

Reply:
This has been updated